parts/windowsProvision.ps1

```diff
@@ -50,115 +298,137 @@ function InstallOpenSSH()
+    Start-Process -Wait -FilePath ${psw_installer} -ArgumentList "--extract-folder $installDir --x"
+    Start-Process -Wait -FilePath "$installDir\setup" -ArgumentList "install --eula=accept --output=$installDir\intel_install.log --components=all"
+}
 
-$is_vanilla = "IS_VANILLA_VM"
-if ($is_vanilla -eq "true") {
-    Write-Output "Skipping Open Enclave installation."
-    exit 0
+function Install-VisualStudio {
+    $installerArguments = @(
+        "-q", "--wait", "--norestart", "--nocache",
+        "--add Microsoft.VisualStudio.Workload.MSBuildTools",
+        "--add Microsoft.VisualStudio.Workload.VCTools",
+        "--add Microsoft.VisualStudio.Component.VC.Tools.x86.x64",
+        "--add Microsoft.VisualStudio.Component.VC.140",
+        "--add Microsoft.VisualStudio.Component.Windows10SDK.16299.Desktop",
+        "--add Microsoft.VisualStudio.Component.Windows81SDK",
+        "--add Microsoft.VisualStudio.Component.VC.ATL"
+    )  
+    Install-CITool -InstallerPath -InstallerPath $PACKAGES["vs_buildtools"]["local_file"] `
+                   -ArgumentList $installerArguments
+   
+    [Environment]::SetEnvironmentVariable("VS150COMNTOOLS", "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2017\BuildTools\Common7\Tools", "Machine")
 }
-Write-Output "Installing Open Enclave"
 
-# Install the intel sgx drivers
-& curl.exe  -o "c:/tmp/sgx_base.cab" "http://download.windowsupdate.com/d/msdownload/update/driver/drvs/2018/01/af564f2c-2bc5-43be-a863-437a5a0008cb_61e7ba0c2e17c87caf4d5d3cdf1f35f6be462b38.cab"
-& 7z x c:/tmp/sgx_base.cab -o"c:/tmp/sgx_base" -y
-&pnputil /add-driver c:/tmp/sgx_base/sgx_base.inf
+function Install-Cmake {
+    $installDir = Join-Path $env:ProgramFiles "CMake"
 
-$psw_uri = "http://registrationcenter-download.intel.com/akdlm/irc_nas/13688/Intel%20SGX%20PSW%20for%20Windows%20v2.1.100.46245.exe"
-$psw_file = "c:/tmp/Intel%20SGX%20PSW%20for%20Windows%20v2.1.100.46245.exe"
+    Install-CITool -InstallerPath $PACKAGES["cmake"]["local_file"] `
+                   -InstallDirectory $installDir `
+                   -ArgumentList @("/quiet", "/passive") `
+                   -EnvironmentPath @("$installDir\bin")
 
-& curl.exe -o $psw_file $psw_uri
+}
 
-& 7z x $psw_file -y
-$psw_installer = " C:\tmp\Intel SGX PSW for Windows v2.1.100.46245\PSW\Intel(R)_SGX_Windows_x64_PSW_2.1.100.46245.exe"
-Start-Process -Wait -FilePath $psw_installer -ArgumentList "--extract-folder c:/tmp/intel_psw_install --x"
-Start-Process -Wait -FilePath "c:/tmp/intel_psw_install/setup" -ArgumentList "install --eula=accept --output=c:/tmp/intel_install.log --components=all"
+function Install-Ocaml {
+    $installDir = Join-Path $env:ProgramFiles "ocpwin64"
+    try {
+        Install-ZipCITool -ZipPath $PACKAGES["ocaml"]["local_file"] `
+                          -InstallDirectory $installDir `
+                          -EnvironmentPath @("$installDir\4.02.1+ocp1-msvc64-20160113\bin")

```

Comment:
We should have OCaml installed directly under `ocpwin64` directory, without the intermediate `4.02.1+ocp1-msvc64-20160113` directory which is a bit hard-coded.